lib/splitclient-rb/cache/stores/localhost_split_store.rb

```diff
@@ -0,0 +1,193 @@
+module SplitIoClient
+  module Cache
+    module Stores
+      class LocalhostSplitStore
+        attr_reader :splits_repository
+
+        def initialize(splits_repository, config, sdk_blocker = nil)
+          @splits_repository = splits_repository
+          @config = config
+          @sdk_blocker = sdk_blocker
+
+          if @config.using_default_split_file?
+            @config.logger.warn('Localhost mode: .split mocks ' \
+              'will be deprecated soon in favor of YAML files, which provide more ' \
+              'targeting power. Take a look in our documentation.')
+          end
+        end
+
+        def call
+          if ENV['SPLITCLIENT_ENV'] == 'test'
+            store_splits
+          else
+            splits_thread
+
+            if defined?(PhusionPassenger)
+              PhusionPassenger.on_event(:starting_worker_process) do |forked|
+                splits_thread if forked
+              end
+            end
+          end
+        end
+
+        private
+
+        def splits_thread
+          @config.threads[:split_store] = Thread.new do
+            @config.logger.info('Starting splits fetcher service')
+            loop do
+              store_splits
+
+              sleep(StoreUtils.random_interval(@config.offline_refresh_rate))
+            end
+          end
+        end
+
+        def store_splits
+          load_features.each do |split|
+            store_split(split)
+          end
+
+          if @sdk_blocker
+            @sdk_blocker.splits_ready!
+            @sdk_blocker.segments_ready!
+          end
+        rescue StandardError => error
+          @config.logger.error('Error while parsing the split file. Check that the input file matches the expected format')
+          @config.log_found_exception(__method__.to_s, error)
+        end
+
+        def store_split(split)
+          @config.logger.debug("storing split (#{split[:name]})") if @config.debug_enabled
+
+          @splits_repository.add_split(split)
+        end
+
+        def load_features
+          yaml_extensions = [".yml", ".yaml"]
+          if yaml_extensions.include? File.extname(@config.split_file)
+            parse_yaml_features
+          else
+            parse_plain_text_features
```

Comment:
I think that we can put the `warn` here and remove the `using_default_split_file` method from the config.  
Putting the log here, we cover all scenarios when the customers use `.split` files.

Reply:
Done! good catch